verification/test-wdls/TestArrays.wdl

```diff
@@ -0,0 +1,295 @@
+                            select_all([Arrays.fingerprint_summary_metrics_file]),
+                            select_all([Arrays.genotype_concordance_summary_metrics_file]),
+                            select_all([Arrays.genotype_concordance_detail_metrics_file]),
+                            select_all([Arrays.genotype_concordance_contingency_metrics_file]),
+    ])
+
+    # Copy results of pipeline to test results bucket
+    call Copy.CopyFilesFromCloudToCloud as CopyToTestResults {
+      input:
+        files_to_copy             = flatten([pipeline_outputs, pipeline_metrics]),
+        vault_token_path          = vault_token_path,
+        google_account_vault_path = google_account_vault_path,
+        destination_cloud_path    = results_path
+    }
+
+    # If updating truth then copy output to truth bucket
+    if (update_truth){
+      call Copy.CopyFilesFromCloudToCloud as CopyToTruth {
+        input: 
+          files_to_copy             = flatten([pipeline_outputs, pipeline_metrics]),
+          vault_token_path          = vault_token_path,
+          google_account_vault_path = google_account_vault_path,
+          destination_cloud_path    = results_path
+      }
+    }
+
+    # If not updating truth then we need to collect all input for the validation WDL
+    # This is achieved by passing each desired file/array[files] to GetValidationInputs
+    if (!update_truth) {
+      call Utilities.GetValidationInputs as GetMetricsInputs {
+        input:
+          input_files  = pipeline_metrics,
+          results_path = results_path,
+          truth_path   = truth_path
+        }
+
+      call Utilities.GetValidationInputs as GetGtcs {
+        input:
+            input_file   = Arrays.gtc_file,
+            results_path = results_path,
+            truth_path   = truth_path
+        }
+  
+      call Utilities.GetValidationInputs as GetVCFs {
+        input:
+          input_file   = Arrays.output_vcf,
+          results_path = results_path,
+          truth_path   = truth_path
+      }
+  
+      call Utilities.GetValidationInputs as GetFingerprintVCFs { 
+        input:
+          input_file   = Arrays.output_fingerprint_vcf,
+          results_path = results_path,
+          truth_path   = truth_path
+      }
+
+      call Utilities.GetValidationInputs as GetRedIdat { 
+        input:
+          input_file   = Arrays.red_idat_md5_cloud_path,
+          results_path = results_path,
+          truth_path   = truth_path
+      }
+
+      call Utilities.GetValidationInputs as GetGreenIdat { 
+        input:
+          input_file   = Arrays.green_idat_md5_cloud_path,
+          results_path = results_path,
+          truth_path   = truth_path
+      }
+
+      call Utilities.GetValidationInputs as GetParams { 
+        input:
+          input_file   = Arrays.chip_well_barcode_params_file,
+          results_path = results_path,
+          truth_path   = truth_path
+      }
+
+      call VerifyArrays.VerifyArrays as VerifyArrays {
```

Comment:
do you need to add the dummy input "done" to force completion before we verify?

Reply:
good catch!